autoload/neomake/utils.vim

```diff
@@ -200,3 +200,20 @@ function! neomake#utils#GetHighlight(group, what) abort
   endif
   return val
 endfunction
+
+function! neomake#utils#ExpandArgs(args) abort
+    if neomake#utils#IsRunningWindows()
+        call neomake#utils#ExpandArgsInWindows(a:args)
+    else
+        call neomake#utils#ExpandArgsInLinux(a:args)
+    endif
+endfunction
+
+function! neomake#utils#ExpandArgsInWindows(args) abort
+    " Don't expand &shellcmdflag argument of cmd.exe nor arguments like '"something"'
+    call map(a:args, "(v:val ==? &shellcmdflag || v:val =~? '.*\".*\".*') ? v:val : expand(v:val)")
+endfunction
+
+function! neomake#utils#ExpandArgsInLinux(args) abort
+    call map(a:args, 'expand(v:val)')
```

Comment:
I have just found a weird issue in this regard, when trying to reproduce something else with the Vim version used on Travis.  On Arch Linux, `expand()` behaves weird:

        " XXX: with locally built Vim 7.3.429 (like on Travis), this expands
        " weird, and results in a command-not-found error:
        " :echo expand("sh -c '1 2 3'")
        " sh
        " -c
        " '1
        " 2
        " 3'
        call map(args, 'expand(v:val)')

I don't know why that does not happen on Travis though?!

I think we should not just use `expand` like that, but only expand words in list items that start with a`%`, or something similar restricted.
This way the hack for Windows would not be required at all?!